packages/runtime/src/util.ts

```diff
@@ -102,3 +102,80 @@ export const construct = supportReflect
     }
     return instance
   }
+
+export const utf8Decoder: { decode: (input: BufferSource) => string } = typeof TextDecoder === 'function'
+  ? new TextDecoder()
+  : {
+      decode (input: BufferSource) {
```

Comment:
I'd suggest to either exclude the fallback, or at least guard it under `#if TEXTDECODER == 2` etc like Emscripten does.

It's quite a bit of code for something that none of the modern engines needs.